pandas/tests/indexing/test_iloc.py

```diff
@@ -118,6 +118,40 @@ def check(result, expected):
         with pytest.raises(IndexError, match=msg):
             dfl.iloc[:, 4]
 
+    def test_iloc_non_integer(self):
+
+        # iloc should throw IndexError if non-numeric list is passed
+        df = DataFrame(np.random.random_sample((20, 5)), columns=list('ABCDE'))
+
+        msg = '.iloc requires integer indexers, got'
+        with pytest.raises(IndexError, match=msg):
```

Comment:
Can you move these into a separate method with `_raises` appended to the name and leverage parametrization?

Reply:
I refactored the tests into `test_iloc_non_integer_raises`. Let me know if that is what you were looking for.